analyzer/protocol/ldap/ldap.spicy

```diff
@@ -0,0 +1,638 @@
+# https://tools.ietf.org/html/rfc4511#section-4.7
+
+type AddRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: add request sequence
+
+  on %done {
+    print "AddRequest", message.messageID, self;
+  }
+};
+
+type AddResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "AddResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Delete Operation
+# https://tools.ietf.org/html/rfc4511#section-4.8
+
+type DelRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  objectName: asn1::ASN1Message(True, self.depth) &convert=$$.body.octetstring;
+
+  on %done {
+    message.obj = self.objectName;
+    print "DelRequest", message.messageID, self;
+  }
+};
+
+type DelResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "DelResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Modify DN Operation
+# https://tools.ietf.org/html/rfc4511#section-4.8
+
+type ModDNRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse mod DN request sequence
+
+  on %done {
+    print "ModDNRequest", message.messageID, self;
+  }
+};
+
+type ModDNResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "ModDNResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Compare Operation
+# https://tools.ietf.org/html/rfc4511#section-4.10
+
+type CompareRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  # TODO: parse compare request entry/ava sequence
+
+  on %done {
+    print "CompareRequest", message.messageID, self;
+  }
+};
+
+type CompareResponse = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+
+  result: Result(self.depth+1);
+
+  on %done {
+    message.result = self.result;
+    print "CompareResponse", message.messageID, self;
+  }
+};
+
+#-----------------------------------------------------------------------------
+# Abandon Operation
+# https://tools.ietf.org/html/rfc4511#section-4.11
+
+type AbandonRequest = unit(inout message: Message) {
+  var depth: uint32 = message.seq.body.seq.submessages[1].depth+1;
+  # messageID: asn1::ASN1Message(True, self.depth) &convert=$$.body.num_value;
```

Comment:
Don't we need to at least skip over the data here?

Reply:
In this case we have already parsed the data, we're just doing a `parse-from` the parent Message. ASN.1 already descended the tree for us. For now I've commented this out with the other unimplemented ones.